Mathlib/NumberTheory/Fermat.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2024 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Ring.Basic
+import Mathlib.Algebra.Order.Star.Basic
+import Mathlib.Data.Nat.Prime.Defs
+import Mathlib.Tactic.Ring.RingNF
+
+/-!
+# Fermat numbers
+
+The Fermat numbers are a sequence of natural numbers defined as $2^{2^n} + 1$, for all natural
+numbers $n$.
+
+## Main theorems
+
+- `coprime_fermat_fermat`: two distinct fermat numbers are coprime.
+-/
+
+open Nat
+open scoped BigOperators
+
+/-- The Fermat numbers:
+the $n$-th Fermat number $F_n$ is defined as $2^{2^n} + 1$. -/
+def fermat (n : ℕ) : ℕ := 2 ^ (2 ^ n) + 1
+
+@[simp] theorem fermat_zero : fermat 0 = 3 := rfl
+@[simp] theorem fermat_one : fermat 1 = 5 := rfl
+@[simp] theorem fermat_two : fermat 2 = 17 := rfl
+
+theorem strictMono_fermat : StrictMono fermat := by
+  apply strictMono_nat_of_lt_succ
+  simp only [fermat, add_lt_add_iff_right, Nat.pow_succ]
+  exact fun n => (pow_lt_pow_iff_right one_lt_two).mpr (by aesop)
+
+theorem two_lt_fermat {n : ℕ} : 2 < fermat n := by
```

Comment:
ok, fixed. 
I had kinda thought that if `fermat n` appears, it would be enough to infer it, but of course I suppose often it might appear not in this form, but just as number.

Reply:
The general rule is that arguments can be implicit if they can be inferred from *other explicit arguments*, not from the type of the conclusion (with an exception for `Iff` lemmas when you can implicit things if they can be inferred from other arguments + the type of either half of the "iff").